ddtrace/contrib/anthropic/patch.py

```diff
@@ -0,0 +1,290 @@
+import json
+import os
+import sys
+from typing import Any
+from typing import Iterable
+from typing import Optional
+from typing import Union
+
+import anthropic
+from pydantic import SecretStr
+
+from ddtrace import config
+from ddtrace.contrib.trace_utils import unwrap
+from ddtrace.contrib.trace_utils import with_traced_module
+from ddtrace.contrib.trace_utils import wrap
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.utils.version import parse_version
+from ddtrace.llmobs._integrations import AnthropicIntegration
+from ddtrace.pin import Pin
+
+
+log = get_logger(__name__)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(anthropic, "__version__", "")
+
+
+ANTHROPIC_VERSION = parse_version(get_version())
+BASE_MODULE = anthropic
+BASE_MODULE_NAME = getattr(anthropic, "__name__", "anthropic")
+
+
+config._add(
+    "anthropic",
+    {
+        "span_prompt_completion_sample_rate": float(os.getenv("DD_ANTHROPIC_SPAN_PROMPT_COMPLETION_SAMPLE_RATE", 1.0)),
+        "span_char_limit": int(os.getenv("DD_ANTHROPIC_SPAN_CHAR_LIMIT", 128)),
+    },
+)
+
+
+def _extract_model_name(instance: Any) -> Optional[str]:
+    """Extract model name or ID from llm instance."""
+    for attr in "model":
+        if hasattr(instance, attr):
+            return getattr(instance, attr)
+    return None
+
+
+def _format_api_key(api_key: Union[str, SecretStr]) -> str:
+    """Obfuscate a given LLM provider API key by returning the last four characters."""
+    if hasattr(api_key, "get_secret_value"):
+        api_key = api_key.get_secret_value()
+
+    if not api_key or len(api_key) < 4:
+        return ""
+    return "...%s" % api_key[-4:]
```

Comment:
Is it deliberate to not use an f-string here?

Reply:
Removed